main_test.go

```diff
@@ -168,9 +155,122 @@ func TestEnd2EndEnvironmentVariablesSuccess(t *testing.T) {
 
 	// Merge files
-	mergeFilesInHierarchy(hierarchy, cfg.filterExtension, cfg.outputFile)
+	mergeFilesInHierarchy(hierarchy, cfg.filterExtension, cfg.outputFile, false, false)
+
+	expected, err := ioutil.ReadFile("testdata/hierarchy-with-env/result/expected.yaml")
+	if err != nil {
+		t.Fatalf("Error reading file with expected test results: %v", err)
+	}
+	result, err := ioutil.ReadFile(cfg.outputFile)
+	if err != nil {
+		t.Fatalf("Error reading output file: %v", err)
+	}
+	assert.Equal(t, string(expected), string(result))
+}
+
+// TestMissingHierarchySuccess runs through the full functionality end-to-end
+// It test the case where no hierarchy.lst is provided,
+// meaning only the base directory is searched for files to merge
+// It compares the generated final file with one stored in git
+func TestMissingHierarchySuccess(t *testing.T) {
+
+	cfg := cfgDefaults
+	cfg.basePath = "testdata/no-hierarchy"
+
+	// process the hierarchy and get the list of include files
+	hierarchy := processHierarchy(cfg)
+
+	// Lets do the deed
+	mergeFilesInHierarchy(hierarchy, cfg.filterExtension, cfg.outputFile, false, false)
+
+	expected, err := ioutil.ReadFile("testdata/no-hierarchy/result/expected.yaml")
+	if err != nil {
+		t.Fatalf("Error reading file with expected test results: %v", err)
+	}
+	result, err := ioutil.ReadFile(cfg.outputFile)
+	if err != nil {
+		t.Fatalf("Error reading output file: %v", err)
+	}
+	assert.Equal(t, string(expected), string(result))
+}
+
+// TestFailMissingHierarchy runs through the full functionality end-to-end
+// It test the case where no hierarchy.lst is provided,
+// meaning only the base directory is searched for files to merge
+// It compares the generated final file with one stored in git
+func TestFailMissingHierarchy(t *testing.T) {
+	if os.Getenv("TEST_FAIL_MISSING_HIERARCHY") == "1" {
+		cfg := cfgDefaults
+		cfg.basePath = "testdata/no-hierarchy"
+		cfg.failMissingHierarchy = true
+
+		processHierarchy(cfg)
+
+		return
+	}
+
+	cmd := exec.Command(os.Args[0], "-test.run=TestFailMissingHierarchy")
+	cmd.Env = append(os.Environ(), "TEST_FAIL_MISSING_HIERARCHY=1")
+	output, err := cmd.CombinedOutput()
+	fmt.Printf("%s\n", output)
+	if e, ok := err.(*exec.ExitError); ok && !e.Success() {
+		fmt.Printf("Process correctly failed with %v\n", e)
+		return
+	}
+	t.Fatalf("process ran with err %v, want exit status 1.", err)
+
+}
+
+// TestFailHierarchyMissingEnvironmentVariable ensures that the application is correctly failing
+// If an environment variable specified in the yaml content is not defined.
+// It spawns a new process to determine the exit code of the application.
+// Anything other than a 1 is a problem
+// It uses the environment variable TEST_FAIL_EMPTY to signal the actual execution of the functionality
+func TestFailContentMissingEnvironmentVariable(t *testing.T) {
+	if os.Getenv("TEST_FAIL_EMPTY") == "1" {
+		cfg := cfgDefaults
+		cfg.basePath = "testdata/content-with-env"
+		cfg.failMissingEnvVar = false
```

Comment:
```suggestion
```
Isn't this already set in cfgDefaults? Also, I would think this should be " cfg.failMissingEnvVar = true"